EIPS/eip-7702.md

```diff
@@ -0,0 +1,97 @@
+
+At the start of executing the transaction, for each `[contract_code, y_parity, r, s]` tuple:
+
+1. Let `signer = ecrecover(keccak(MAGIC + contract_code), y_parity, r, s]`.
+2. Verify that the contract code of `signer` is empty.
+3. Set the contract code of `signer` to `contract_code`.
+
+At the end of the transaction, set the `contract_code` of each `signer` back to empty.
+
+Note that the signer of any of the `contract_code` signatures, and the `tx.origin` of the transaction, are allowed to be different.
+
+## Rationale
+
+### Conversion of EIP-3074 use cases
+
+In this design, it requires fairly little work to convert an existing EIP-3074 workflow. Specifically, AUTH and AUTHCALL would get replaced by calls into the EOA. One way to do this is that the `contract_code` would be a user wallet (which could be a `DELEGATECALL` forwarder to save gas), and would expose two functions, `verify` and `execute`.
+
+* AUTH would be replaced by a code to `verify`, which would use TSTORE to locally set `authorized[msg.sender, ...] = True`.
+* AUTHCALL would be replaced by a call to `execute`, which would use TLOAD to verify `authorized[msg.sender, ...]`, and then execute from there.
+
+Hence, there is a very simple transformation from "existing EIP-3074 workflows" into workflows under this new scheme.
+
+### Forward-compatibility with future account abstraction
+
+This EIP is designed to be very forward-compatible with endgame account abstraction, without over-enshrining any fine-grained details of [ERC-4337](./eip-4337.md) or RIP-7560.
+
+Specifically:
+
+* The contract code that users would need to sign could literally be existing ERC-4337 wallet code.
+* The "code pathways" that are used are code pathways that would, in many cases (though perhaps not all), continue to "make sense" in a pure-smart-contract-wallet world.
+* Hence, it avoids the problem of "creating two separate code ecosystems", because to a large extent they would be the same ecosystem. There would be some workflows that require kludges under this solution that would be better done in some different "more native" under "endgame AA", but this is relatively a small subset.
+* It does not require adding any opcodes, that would become dangling and useless in a post-EOA world.
+* It allows EOAs to temporarily convert themselves into contracts to be included in ERC-4337 bundles, in a way that's compatible with the existing `EntryPoint`.
+* Once this is implemented, [EIP-5003](./eip-5003.md) is "only one line of code": just add a flag to not set the code back to empty at the end.
+
+## Backwards Compatibility
+
+This EIP breaks the invariant that an account balance can only decrease as a result of transactions originating from that account. This has consequences for mempool design, and for other EIPs such as inclusion lists. However, these issues are common to any proposal that provides similar functionality, including EIP-3074.
+
+## Security Considerations
```

Comment:
I assume it is, but just to confirm - this EIP doesn't allow storage modification of the EOA right?
e.g., SSTORE, etc

Reply:
https://github.com/ethereum/EIPs/pull/8527/files#r1593099283